File: Machine Learning A-Z Template Folder/Part 2 - Regression/Section 9 - Random Forest Regression/Exercises/random_forest_regression.r
# Random Forest Regression

library(e1071)
library(randomForest)
library(ggplot2)

#-------------------------------------------------------------------------------
#                            Loading in Data
#-------------------------------------------------------------------------------

dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]
print('-------------------------------------------------------------------')
print(dataset)
print('-------------------------------------------------------------------')
summary(dataset)
print('-------------------------------------------------------------------')

#-------------------------------------------------------------------------------
#                 Fitting Random Forest Regression to the dataset
#-------------------------------------------------------------------------------

set.seed(1234)
regressor = randomForest(x = dataset[1],
                         y = dataset$Salary,
                         ntree = 500)
summary(regressor)

#-------------------------------------------------------------------------------
#             Predicting values for Decision Tree Regression
#-------------------------------------------------------------------------------

y_pred = predict(regressor, data.frame(Level = 6.5))
print("=======================================")
print(y_pred)
print("=======================================")

#-------------------------------------------------------------------------------
#             Visualizing Random Forest Regression model to the dataset
#-------------------------------------------------------------------------------

png('random_forest_regression.png')
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.01)
random_forest_plot = ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary), colour = 'red') +
  geom_line(aes(x = x_grid, y = predict(regressor, newdata = data.frame(Level=x_grid))), colour = 'blue') +
  ggtitle('Salary prediction (Random Forest Regression)') +
  xlab('Level') +
  ylab('Salary')
plot(random_forest_plot)
dev.off()

File: Machine Learning A-Z Template Folder/Part 2 - Regression/Section 9 - Random Forest Regression/Exercises/random_forest_regression.py
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.ensemble import RandomForestRegressor
from sklearn.preprocessing import StandardScaler

#-------------------------------------------------------------------------------
#                            Loading in Data
#-------------------------------------------------------------------------------

dataset = pd.read_csv('Position_Salaries.csv')
print('\n-------------------------------------------------------------------\n')
print("Head of dataset:\n",dataset.head())
print('\n-------------------------------------------------------------------\n')
x = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values
print("x:\n", x)
print('\n-------------------------------------------------------------------\n')
print("y:\n",y)
print('\n-------------------------------------------------------------------\n')

#-------------------------------------------------------------------------------
#                  Creating Random Forest Regressor
#-------------------------------------------------------------------------------

regressor = RandomForestRegressor(n_estimators = 1000, random_state = 0)
regressor.fit(x, y)
pred = regressor.predict(6.5)
print("prediction of salary for 6.5 years of experience: ", pred)
print('\n-------------------------------------------------------------------\n')

#-------------------------------------------------------------------------------
#                  Visualizing Random Forest Regression
#-------------------------------------------------------------------------------
x_grid = np.arange(min(x), max(x), 0.001)
x_grid = x_grid.reshape((len(x_grid), 1))

plt.scatter(x, y, color = 'r')
plt.plot(x_grid, regressor.predict(x_grid), c = 'blue')
plt.title("Truth or Bluff (Random Forest Regression)")
plt.xlabel('Position level')
plt.ylabel('Salary')
plt.show()
